tools/language-server/logstat/src/main.rs

```diff
@@ -0,0 +1,317 @@
+
+    while let Some(line) = lines.next_line().await? {
+        if let Some(cap) = RE_LOG.captures(line.as_str()) {
+            let groups = (cap.get(2), cap.get(3));
+            match groups {
+                (Some(timestamp), Some(message)) =>
+                    if message.as_str().contains(current_match) {
+                        let duration = Duration::ZERO;
+                        let timestamp =
+                            OffsetDateTime::parse(timestamp.as_str(), &Rfc3339).unwrap();
+                        let line = message.as_str().to_string();
+
+                        current_operations.push(Operation { duration, timestamp, line });
+
+                        if let Some(m) = matches.next() {
+                            current_match = m
+                        } else {
+                            matches = spec.matches.iter();
+                            current_match = matches.next().expect("Empty spec!");
+                            iterations.push(Iteration { operations: current_operations });
+                            current_operations = vec![];
+                        }
+                    },
+                _ => {
+                    eprintln!("[ERR] Invalid log line [{}]", line);
+                }
+            }
+        }
+    }
+
+    Ok(iterations)
+}
+
+
+/// Calcualte median of values
+fn median(durations: &mut [Duration]) -> Duration {
+    durations.sort();
+    let mid = durations.len() / 2;
+    if durations.len() % 2 == 0 {
+        (durations[mid - 1] + durations[mid]) / 2
+    } else {
+        durations[mid]
+    }
+}
+
+/// Cleanup iterations info before analyzing
+fn cleanse_iterations(ops: &mut Vec<Iteration>, skip: usize) {
+    ops.drain(..skip);
+}
+
+/// Merge iterations from two log files
+fn merge_iterations(ws_iterations: &mut Vec<Iteration>, log_iterations: Vec<Iteration>) {
+    for (log_iter, ws_iter) in log_iterations.into_iter().zip(ws_iterations.into_iter()) {
+        ws_iter.operations.splice(1..1, log_iter.operations);
+    }
+}
+
+/// Calculate operation durations for each benchmark iteration
+fn calculate_durations(iterations: &mut Vec<Iteration>) {
+    for iteration in iterations {
+        let mut timestamp = OffsetDateTime::UNIX_EPOCH;
+
+        for mut op in iteration.operations.iter_mut() {
+            if timestamp == OffsetDateTime::UNIX_EPOCH {
+                timestamp = op.timestamp
+            }
+            op.duration = op.timestamp - timestamp;
+            timestamp = op.timestamp;
+        }
+    }
+}
+
+/// Analyze benchmark results
+fn analyze_iterations(iterations: &[Iteration], use_median: bool) -> Vec<Stats> {
+    let mut stats = vec![];
+
+    let ops_len = iterations.len();
+    let splits_len = iterations.first().unwrap().operations.len();
+
+    let mut i = 0;
+    while i < splits_len {
+        let mut min = Duration::MAX;
+        let mut max = Duration::ZERO;
+        let mut sum = Duration::ZERO;
+        let mut durations = vec![];
```

Comment:
Another place where we can just 
```rust
let durations = iterations.map(|it| it.operations[i].duration).collect();
```
And for min, max and sum, you can use appropriate method of `Iterator`, for example [min](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.min).